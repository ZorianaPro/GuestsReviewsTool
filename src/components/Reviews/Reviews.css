.Reviews {
  z-index: 2;
  width: calc(100% - 16em);
  position: relative;
  display: block;
}

@media only screen and (max-width: 992px) {
  .Reviews {
    width: 100%;
  }
}