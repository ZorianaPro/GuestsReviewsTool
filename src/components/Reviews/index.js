import React from 'react';
import Review from '../Review';
import Paper from '../Paper';
import Loading from '../Loading';
import './Reviews.css';
import LoadMore from '../LoadMore';

const Reviews = ({
  reviews,
  total,
  isLoading,
  shouldShowLoadButton,
  onCLickLoadMore
}) => (
  <div className="Reviews">
    <Paper>
      {
        total !== 0
        && <h1>{ total } Reviews</h1>
      }
      {
        !total && isLoading
        && <Loading/>
      }
      {
        !total && !isLoading
        && <h1>Sorry, no results...</h1>
      }
      {
        (reviews || [])
          .map((review) =>
            <Review
              score={ review.score }
              author={ review.author }
              headline={ review.headline }
              comment={ review.comment }
              positiveFeedback={ review.positiveFeedback }
              negativeFeedback={ review.negativeFeedback }
              channel={ review.channel }
              publishedAt={ review.publishedAt }
            />
          )
      }
    </Paper>
    {
      shouldShowLoadButton
      && <LoadMore
        onClick={
          onCLickLoadMore
        }
        isLoading={
          isLoading
        }/>
    }
  </div>
);

export default Reviews;