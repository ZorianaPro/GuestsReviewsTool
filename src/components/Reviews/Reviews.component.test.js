import React from 'react';
import { shallow, mount } from 'enzyme';
import Reviews from './';

describe('Reviews', () => {
  it('renders without crashing', () => {
    expect(
      shallow.bind(
        shallow,
        <Reviews/>
      ))
      .not
      .toThrow();
  });

  it('renders as expected', () => {
    expect(
      mount(
        <Reviews />
      )
    ).toMatchSnapshot();
  });

  describe('if given total prop', () => {
    let reviews;
    let total;

    beforeEach(() => {
      reviews = mount(<Reviews total='2'/>);
      total = reviews.findWhere((node) => (
        node.name() === 'h1'
        && node.text() === '2 Reviews'
      ));
    });

    it('renders total', () => {
      expect(total.exists())
        .toBe(true);
    });
  });

  describe('if given reviews', () => {
    let reviews;
    const reviewsList = [
      'one',
      'two'
    ];

    beforeEach(() => {
      reviews = mount(<Reviews reviews={ reviewsList }/>);
    });

    it('renders reviews', () => {
      expect(reviews.find('.Review').length)
        .toBe(reviewsList.length);
    });
  });
});
