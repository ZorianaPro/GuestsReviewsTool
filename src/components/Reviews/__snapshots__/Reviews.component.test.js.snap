// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews renders as expected 1`] = `
<Reviews>
  <div
    className="Reviews"
  >
    <Paper>
      <div
        className="Paper"
      >
        <h1>
           Reviews
        </h1>
        <h1>
          Sorry, no results...
        </h1>
      </div>
    </Paper>
  </div>
</Reviews>
`;
